### DISCUSSION TITLE
Run of automake leads to build error of ww3_gint WW3 Version 7.14 - subroutine W3IORS

### DISCUSSION THREAD

**Discussion by PellTheTabber1** [NOAA-EMC/WW3#1108] [Q&A] [answered]
Switches: NOGRB TRKNC SHRD PR1 FLX0 LN1 ST4 STAB0 NL1 BT4 DB1 MLIM TR0 BS1 IC0 IS0 REF1 WNT2 WNX1 RWND CRT1 CRX1 TIDE O0 O1 O2 O2a O2b O2c O3 O4 O5 O6 O7
I get the same basic failure on module ww3_gint with Franklin Ayala's switches from version 6.07 tutorial he published back in 2022:
F90 NOGRB NC4 SHRD PR3 UQ FLX0 LN1 ST4 NL1 BT1 DB1 TR0 BS0 IC0 IS0 REF0 XX0 WNT1 WNX1 CRT1 CRX1 O0 O1 O2 O3 O4 O5 O6 O7 O10 O11

I can compile past ww3_gint but most will not compile without ww3_gint.  Seems it's needed by a number of other programs.  I guess I'll try to test the routines that I've build so far while I'm waiting for some help from someone.  I can build 12 out of the 15 routines as ww3_ounf, ww3_ounp and I think one other depend on ww3_gint to build.  So I guess I'll just use the routines that DO build until I can figure those 3 out.  So I can build 12 of the routines.

This might be my fault but the make was building other modules successfully, but then was unable to link due to the call of a subroutine W3IORS that is undefined.  Is the build missing something or am I missing something?

Also, while I'm at it, this is of less importance, the script that pulls the binary files and large files (NOT in git) down via ftp, deletes the files if you want to save them.  The 'n' answers work opposite to what is expected.  Just thought I'd mention this.

My build error is summarized below - I am looking for how to proceed to get the unresolved reference resolved somehow. I will keep looking myself.

Processing ww3_trck
---------------------
make: '/home/jdbpcix1/umr-lops/WW3/model/exe/ww3_trck' is up to date.
 
Processing ww3_grib
---------------------
make: '/home/jdbpcix1/umr-lops/WW3/model/exe/ww3_grib' is up to date.
 
Processing ww3_gint
---------------------
      Linking ww3_gint
makefile:524: recipe for target '/home/jdbpcix1/umr-lops/WW3/model/exe/ww3_gint' failed
make: *** [/home/jdbpcix1/umr-lops/WW3/model/exe/ww3_gint] Error 1

ww3_gint.o: In function `MAIN__':
/home/jdbpcix1/umr-lops/WW3/model/src/ww3_gint.F90:916: undefined reference to `__w3iorsmd_MOD_w3iors'
ww3_gint.o: In function `w3exgi.3381':
/home/jdbpcix1/umr-lops/WW3/model/src/ww3_gint.F90:3562: undefined reference to `__w3iorsmd_MOD_w3iors'
collect2: error: ld returned 1 exit status

The above-mentioned lines 916:
!To use an older model version restart file (add a w3iorsold)
      !CALL W3IORSOLD ( 'READ', 56, XXX, INTYPE, IG )
      CALL W3IORS ( 'READ', 56, XXX, IG )

 and 3562:      
 CALL W3IORS ( 'HOT', 55, XXX, NG)

**Comment by JessicaMeixner-NOAA** [comment, marked as answer]
@PellTheTabber1 it looks like you're using the latest version of develop but the gnu build system which is not necessarily supported anymore.  It is possible that the gnu build system needs some updates for this to work.  Have you tried CMAKE?

**Reply by PellTheTabber1** [reply]
In other words, CMAKE would literally find an unresolved external reference to a routine where the gnu build didn't resolve it?  Ok, I'll try CMAKE I guess, and maybe update my gnu compiler stuff, thank you.  I've just installed cmake 3.5.1, we'll give it a go.  Update: cmake 3.5.1 is earlier than the minimum cmake which is 3.19.  Could be a problem for me.  I may need to debug the gnu build.  Hard to say if I can upgrade as I'm running Linux Mint 18.3.  I may need to buy a new workstation for this.

Just a comment: I don't mind troubleshooting this a little so I'll try to find where that subroutine lives and modify the makefile in model/src to include it in the build.  It's probably just a .o file that is not listed in the make.  That MIGHT be the problem although I would need to find where that function call to W3IORS.  ww3_gint is not a critically important file but I'd like to see it build.

**Reply by JessicaMeixner-NOAA** [reply]
The GNU build is no longer tested or maintained, so it's likely it's gotten out of date.  CMAKE is tested and should work.

**Reply by PellTheTabber1** [reply]
Just noticed that the file makefile in model/src does not have the required line to compile in module W3IORSMD which is in file w3iorsmd.f90.  Since I have no choice at this point, I will simply make modifications to 2 places in that file where such a line is needed and then try to build again.  It should build ok once that line (similar to this one in ww3_strt: $(aPe)/ww3_strt : $(aPo)/w3iorsmd.o).  I have nothing to do until I set up a new PC, so I might as well troubleshoot a little even if it's not a hugely productive task.  At least I know WHY it's not building (other than processes that are beyond my control).
